vm/src/obj/objfloat.rs

```diff
@@ -134,6 +134,17 @@ fn inner_divmod(v1: f64, v2: f64, vm: &VirtualMachine) -> PyResult<(f64, f64)> {
     }
 }
 
+fn inner_pow(v1: f64, v2: f64, vm: &VirtualMachine) -> PyResult {
+    if v1.is_zero() {
+        let msg = format!("{} cannot be raised to a negative power", v1);
+        Err(vm.new_zero_division_error(msg))
+    } else if (v1 < 0.0) & (v2.floor() != v2) {
```

Comment:
I am not sure `&` is inteded or not. Is this intended to be like below?
```suggestion
    } else if v1.is_negative() && v2.floor() != v2 {
```